Slim/Control/Commands.pm

```diff
@@ -3404,6 +3402,19 @@ sub _playlistXtracksCommand_parseSearchTerms {
 		}
 	}
 
+	# Check for 'sort' after iterating other keys, so that it takes precedence...
+	if (defined($terms->{'sort'})) {
```

Comment:
There's probably still an issue: if you look further up the code, you'd see that `$sort` is only defined under certain circumstances. I believe we should only apply the sort under the same conditions. Eg. check here whether any sort order has been defined, or whether it's identical to `$albumSort`. And only under this condition you'd set it. Otherwise I could imagine some relations might not have been set up.

Reply:
Update code to check if sort was albumSort before changing.